Sources/Core/Selection/FilterSelectionStore.swift

```diff
@@ -5,6 +5,13 @@
 import Foundation
 
 final class FilterSelectionStore {
+
+    // MARK: - Internal properties
+
+    internal var hasChanges = false
```

Comment:
Don't really like this flag here. How about adding a delegate or observer to `FilterSelectionStore`? Then in `CharcoalViewController` you can subscribe to changes and call appropriate `CharcoalViewControllerDelegate` method. There still could be a boolean flag in `CharcoalViewController` to call its delegate only when the user comes back to the root and so on.

Reply:
Sounds good :)

Reply:
It might be better now :P